framework/src/main/java/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -76,6 +80,32 @@ protected void commonAssignmentCheck(
                     getTypeFactory().createIntRangeAnnotation(Range.CHAR_EVERYTHING));
         }
 
+        if (valueTree.getKind() == Kind.METHOD_INVOCATION
+                && TreeUtils.isArrayscopyOfMethodInvocation((MethodInvocationTree) valueTree)
+                && valueType.getKind() == TypeKind.ARRAY) {
+            List<? extends ExpressionTree> args = ((MethodInvocationTree) valueTree).getArguments();
```

Comment:
Please add a check here that the length of `args` is what you expect (i.e. 2) and throw a `BugInCF` otherwise